moabb/evaluations/metasplitters.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+from sklearn.model_selection import (
+    BaseCrossValidator,
+    LeaveOneGroupOut,
+    StratifiedKFold,
+    StratifiedShuffleSplit,
+)
+
+
+class OfflineSplit(BaseCrossValidator):
+
+    def __init__(self, n_folds):
+        self.n_folds = n_folds
+
+    def get_n_splits(self, metadata):
+        subjects = len(metadata.subject.unique())
+        sessions = len(metadata.session.unique())
+        return subjects * sessions
+
+    def split(self, X, y, metadata, **kwargs):
+
+        subjects = metadata.subject
+
+        for subject in subjects.unique():
+            X_, y_, meta_ = (
+                X[subjects == subject],
+                y[subjects == subject],
+                metadata[subjects == subject],
+            )
+            sessions = meta_.session.values
```

Comment:
You don't need to touch the X or the y data; everything here should work only with indices